x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/InferencePyTorchAction.java

```diff
@@ -84,12 +84,13 @@ protected void doRun() throws Exception {
             logger.debug(() -> format("[%s] skipping inference on request [%s] as it has timed out", getDeploymentId(), getRequestId()));
             return;
         }
+        final String requestIdStr = String.valueOf(getRequestId());
         if (isCancelled()) {
+            getProcessContext().getTimeoutCount().incrementAndGet();
+            getProcessContext().getResultProcessor().ignoreResponseWithoutNotifying(requestIdStr);
```

Comment:
Right now, cancelled requests are not counted at all. This leads incorrect total inference request counts, which breaks adaptive allocations. As discussed offline, let's count cancelled requests as timeouts.